be/src/vec/functions/function_cast.h

```diff
@@ -856,12 +856,19 @@ struct ConvertImplFromJsonb {
                     res[i] = 0;
                     continue;
                 }
-
                 if constexpr (type_index == TypeIndex::UInt8) {
+                    // cast from json value to boolean type
                     if (value->isTrue()) {
                         res[i] = 1;
                     } else if (value->isFalse()) {
                         res[i] = 0;
+                    } else if (value->isInt()) {
+                        res[i] = ((const JsonbIntVal*)value)->val() > 0 ? 1 : 0;
```

Comment:
It more common to use `val() == 0 ? 0 : 1`. And `CAST(-1 AS BOOLEAN) ` return 1 in Doris.

Reply:
done